### PR TITLE
[GLT-4261] TAT trend report table metrics

### PR DESCRIPTION
**Pull request by wuall826**
Jira ticket or GitHub issue:

- [x] Includes a change file
- [ ] Updated tests (or n/a)
- [ ] Updated documentation (or n/a)
- [ ] Discussed with CGI (or n/a; see release procedure on wiki for applicability)

### CODE REVIEW

**Comment by wuall826** [comment]
still need to stick "Assay" and "Gate" column, will add another commit. also, would it be better if I use the backend java csv download (like for other reports) for the metric download in this?

File: ts/tat-trend.ts [hunk 1]
```diff
@@ -371,6 +433,174 @@ function updatePlotWithLegend(
...
+    { title: "", colspan: 1 },
+    ...sortedTimeRanges.map((timeRange) => ({
+      title: timeRange,
+      colspan: 3,
+    })),
+  ];
+  const caseTatTableDefinition: TableDefinition<AssayMetrics, void> = {
+    generateColumns: () => dynamicColumns,
+    parentHeaders, // pass parent headers for multi-level header support
+    disablePageControls: true,
```
Comment (djcooke) [inline ts/tat-trend.ts:462]: You've set sorting options for a couple columns, but this removes the sorting function

File: ts/tat-trend.ts [hunk 2]
```diff
@@ -371,6 +433,174 @@ function updatePlotWithLegend(
...
+    filters: [],
+    staticActions: [],
```
Comment (djcooke) [inline ts/tat-trend.ts:464]: These are both optional. Exclude them rather than assigning an empty array

**Review by avarsava — APPROVED** [review]

File: ts/tat-trend.ts [hunk 3]
```diff
@@ -597,13 +601,22 @@ function buildTableFromMetrics(
...
           assayMetrics.timeRanges.push({
             group: timeRange,
-            avgDays: averageDays,
-            medianDays,
+            avgDays:
+              averageDays !== undefined
+                ? averageDays.toString()
+                : NOT_AVAILABLE,
+            medianDays:
+              medianDays !== undefined ? medianDays.toString() : NOT_AVAILABLE,
```
Comment (djcooke) [inline ts/tat-trend.ts:619]: You're setting these values to undefined and then changing undefined to N/A right below. Either store as undefined in the time range (and handle that appropriately elsewhere), or set these to N/A initially.

File: ts/component/table-builder.ts [hunk 4]
```diff
@@ -708,29 +718,33 @@ export class TableBuilder<ParentType, ChildType> {
...
-        this.selectAllCheckbox = undefined;
       }
       this.columns.forEach((column, i) => {
         addColumnHeader(
           row,
           column.title,
-          i == 0 && !this.definition.bulkActions,
+          i === 0 && !this.definition.bulkActions,
           column.headingClass
         );
```
Comment (djcooke) [inline ts/component/table-builder.ts:749]: This is all duplicated. The only difference is the last two parameters of `addColumnHeader`

**Review by djcooke — APPROVED** [review]

**Pull request merged**